tests/io/test_text.py

```diff
@@ -61,3 +62,24 @@ def test_parse_wpc_surface_bulletin():
                                                 [-102, 32], [-103, 31]])
 
     assert all(df.valid == datetime(2021, 6, 28, 18, 0, 0))
+
+
+def test_negative_lat_highres():
+    """Test decoding of high res coordinates with negative latitude."""
+    coord = '-4920344'
+    lon, lat = _decode_coords(coord)
+    assert lon == -034.4
+    assert lat == -49.2
+
+
+def test_negative_lat():
+    """Test decoding of coordinates with negative latitude."""
+    coord_4_digit = '-2378'
+    lon, lat = _decode_coords(coord_4_digit)
+    assert lon == -78
+    assert lat == -23
+
+    coord_5_digit = '-65134'
+    lon, lat = _decode_coords(coord_5_digit)
+    assert lon == -134
+    assert lat == -65
```

Comment:
I would prefer not to directly test a private function, since we can't then easily refactor `parse_wpc_surface_buelletin` (i.e. replacing `_decode_coords`) without massive changes to the tests. Can we instead use `parse_wpc_surface_bulletin` with a small sample analysis line/file and `StringIO`? Something like:
```python
from io import StringIO

def  test_negative_lat():
    sample = StringIO('''
178 
ASUS02 KWBC 281800
CODSUS
 
CODED SURFACE FRONTAL POSITIONS
NWS WEATHER PREDICTION CENTER COLLEGE PARK MD
342 PM EDT MON JUN 28 2021
 
VALID 062818Z
HIGHS 1022 -3961069 1020 -3851069 1026 3750773 1022 4430845 1019 5520728 1018
''')
    df = parse_wpc_surface_bulletin(sample)
    assert df.feature[0] == Point([-106.9, -39.6])
    ...
```

Reply:
Tests changed